File: src/main/java/com/github/sorusclient/client/module/impl/oldanimations/OldAnimations.java
package com.github.sorusclient.client.module.impl.oldanimations;

import com.github.sorusclient.client.module.ModuleDisableable;
import com.github.sorusclient.client.setting.Setting;
import com.github.sorusclient.client.setting.ConfigurableData;

import java.util.List;

public class OldAnimations extends ModuleDisableable {

    private final Setting<Boolean> oldBlockHit;
    private final Setting<Boolean> showArmorDamage;

    public OldAnimations() {
        super("oldAnimations");

        this.register("oldBlockHit", this.oldBlockHit = new Setting<>(false));
        this.register("showArmorDamage", this.showArmorDamage = new Setting<>(false));
    }

    public boolean isOldBlockHit() {
        return this.oldBlockHit.getValue();
    }

    public boolean showArmorDamage() {
        return this.showArmorDamage.getValue();
    }

    @Override
    public void addSettings(List<ConfigurableData> settings) {
        super.addSettings(settings);
        settings.add(new ConfigurableData.Toggle("Old Blockhit", this.oldBlockHit));
        settings.add(new ConfigurableData.Toggle("Show Armor Damage", this.showArmorDamage));
    }

}

File: src/main/kotlin/com/github/sorusclient/client/module/impl/oldanimations/OldAnimations.kt
package com.github.sorusclient.client.module.impl.oldanimations

import com.github.sorusclient.client.module.ModuleDisableable
import com.github.sorusclient.client.setting.ConfigurableData
import com.github.sorusclient.client.setting.ConfigurableData.Toggle
import com.github.sorusclient.client.setting.Setting

class OldAnimations : ModuleDisableable("oldAnimations") {

    private var oldBlockHit: Setting<Boolean>
    private var showArmorDamage: Setting<Boolean>

    init {
        register("oldBlockHit", Setting(false).also { oldBlockHit = it })
        register("showArmorDamage", Setting(false).also { showArmorDamage = it })
    }

    fun isOldBlockHitValue(): Boolean {
        return oldBlockHit!!.value
    }

    fun showArmorDamageValue(): Boolean {
        return showArmorDamage!!.value
    }

    override fun addSettings(settings: MutableList<ConfigurableData>) {
        super.addSettings(settings)
        settings.add(Toggle("Old Blockhit", oldBlockHit))
        settings.add(Toggle("Show Armor Damage", showArmorDamage))
    }

}